### DISCUSSION TITLE
atomWithStorage hydrated value not present in a store

### DISCUSSION THREAD

**Discussion by grabbou** [pmndrs/jotai#1745] [Ideas]
Hey!

When using an external store and atomWithStorage, its initial value is undefined when accessed via `store.get`. When accessed directly via `useAtomValue(anAtomWithStorage)`, the value is there.

Is this expected behavior? Studying the source code, I can see that updates happen explicitly for any set/hydrate operation run from within React context.

**Comment by dai-shi** [comment]
You are probably reading from a different store.
I should have noted that the store api is for Jotai expert users, at least for now.

**Reply by grabbou** [reply]
I only have one store and it's passed to a top-level Provider. I would like to note that the same code works once `anAtomWithStorage` value has been explicitly set via `set` action.

Here's use-case:
- accessToken is an `atomWithStorage` and is read from outside of React context via `store.get`
- on initial load, atomWithStorage value is read from AsyncStorage and inside React via useAtomValue *is defined*, but via *store.get* is undefined
- logout and then login so that accessToken is explicitly set via `set` action from React context
- run API call - accessToken can be retrieved via `store.get` and is defined

**Comment by dai-shi** [comment]
To get more help, I think you would need to create a minimal reproduction with https://codesandbox.io/s/react-new, without the notion of accessToken and with localStorage (web) instead of AsyncStorage (RN).
